cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -401,7 +401,7 @@ struct ARROW_EXPORT Aggregate {
   std::string function;
 
   /// options for the aggregation function
-  const FunctionOptions* options;
+  std::shared_ptr<FunctionOptions> options;
```

Comment:
is `unique_ptr` possible?

Reply:
Good point, initially the idea was to put a `unique_ptr`, but there was a discussion in the JIRA and finally settled for the `shared_ptr`.

Reply:
Ah, I didn't see that.

Hmm, I suppose it's not much of a difference here.